src/plugins/discord/fetchDiscord.js

```diff
@@ -0,0 +1,190 @@
+ * Returns either the response.json or rejects with
+ * a retry flag and an optional message.
+ *
+ */
+function tryDiscordFetch(fetch, url, fetchOptions): Promise<any> {
+  return fetch(url, fetchOptions).then((response) => {
+    switch (response.status) {
+      case 403:
+        return Promise.reject({
+          retry: false,
+          message: `403 Forbidden: bad API username or key?\n${response.url}`,
+        });
+      case 404:
+        return Promise.reject({
+          retry: false,
+          message: `404 Not Found on: ${response.url}; maybe bad serverUrl?`,
+        });
+      case 410:
+        return Promise.reject({retry: false, message: `410 Gone`});
+      case 429:
+        return Promise.reject({retry: true, message: null});
+      default:
+        return Promise.resolve(response.json());
+    }
+  });
+}
+
+/**
+ * Retries fetch if retry = true in rejected tryDiscordFetch response.
+ * Throws an error if message provided in rejected tryDiscordFetch response.
+ *
+ */
+function retryDiscordFetch(fetch, url, fetchOptions) {
+  return new Promise((resolve, reject) => {
+    const operation = retry.operation();
+    operation.attempt(() => {
+      tryDiscordFetch(fetch, url, fetchOptions)
+        .then((result) => {
+          resolve(result);
+        })
+        .catch((error) => {
+          if (error.retry && operation.retry(true)) {
+            return;
+          } else if (error.message) {
+            reject(new Error(error.message));
+          } else {
+            reject(error);
+          }
+        });
+    });
+  });
+}
+
+/**
+ * Returns a function that ties the fetch options and
+ * Discord server url to the supplied endpoint.
+ * Clients can provide bot token here to create a Discord Fetch.
+ */
+export function makeDiscordFetch(fetch: typeof fetch, token: BotToken) {
+  return async (endpoint: string) => {
+    const fetchOptions = {
+      method: "GET",
+      headers: {
+        Accept: "application/json",
+        Authorization: `Bot ${token}`,
+      },
+    };
+    const url = new URL(`${DISCORD_SERVER}/${endpoint}`).href;
+    return retryDiscordFetch(fetch, url, fetchOptions);
+  };
+}
+
+/**
+ * Iteratively calls provided `fetch` for as long as the `hasNextPage`
+ * variable in the `pageInfo` response is true.
+ *
+ */
+
+export async function paginatedFetch(
+  fetch: (endCursor: EndCursor) => Promise<PageInfo>,
+  endCursor: EndCursor
+): Promise<void> {
+  let after = endCursor;
+  let cont = true;
+
+  while (cont) {
+    const {hasNextPage, endCursor} = await fetch(after);
+    after = endCursor;
+    cont = hasNextPage;
+  }
+}
+
+// Verifies that the provided bot credentials have access to the provided Guild
+export async function validateGuildId(fetcher: Fetcher) {
+  const guildId = fetcher._options.guildId;
+  const guilds = await fetcher.guilds();
+  const ids = guilds.map((x) => x.id);
+  for (const id of ids) {
+    if (id === guildId) {
+      return guildId;
+    }
+  }
```

Comment:
Nit: Looking at this bit again there's a couple of built-in functions this could be simplified with.
[`.some()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) seems to be what you need.
`.map().includes()`
`.map().find()`

`if(guilds.some((x) => x.id === guildId)`

Reply:
Obsoleted